src/background/cookie-instrument.ts

```diff
@@ -1,87 +1,100 @@
-        update.is_session = loggingDB.boolToInt(cookie.isSession);
+        // TODO: Support other cookie operations
+        if (
+          change === "deleted" ||
+          change === "added" ||
+          change === "changed"
+        ) {
+          const update = {} as JavascriptCookieChange;
+
+          update.change = change;
+          // TODO: Add changeInfo.cause
+          update.crawl_id = crawlID;
+
+          const cookie: Cookie = changeInfo.cookie;
 
-        // Accessed time (in microseconds)
-        const lastAccessedTime = new Date(cookie.lastAccessed / 1000); // requires milliseconds
-        update.last_accessed = lastAccessedTime.toLocaleFormat(
-          "%Y-%m-%d %H:%M:%S",
-        );
-        update.raw_host = loggingDB.escapeString(cookie.rawHost);
+          // Creation time (in microseconds)
+          // const creationTime = new Date(cookie.creationTime / 1000); // requires milliseconds
+          // update.creationTime = creationTime.toLocaleFormat("%Y-%m-%d %H:%M:%S");
 
-        cookie = cookie.QueryInterface(Ci.nsICookie);
-        update.expires = cookie.expires;
-        update.host = loggingDB.escapeString(cookie.host);
-        update.is_domain = loggingDB.boolToInt(cookie.isDomain);
-        update.is_secure = loggingDB.boolToInt(cookie.isSecure);
-        update.name = loggingDB.escapeString(cookie.name);
-        update.path = loggingDB.escapeString(cookie.path);
-        update.policy = cookie.policy;
-        update.status = cookie.status;
-        update.value = loggingDB.escapeString(cookie.value);
+          // Expiry time (in seconds)
+          // A comment from pre-webextension code, which may still be valid:
+          // May return ~Max(int64). I believe this is a session
+          // cookie which doesn't expire. Sessions cookies with
+          // non-max expiry time expire after session or at expiry.
+          const expiryTime = cookie.expirationDate; // returns seconds
+          let expiryTimeString;
+          if (!cookie.expirationDate || expiryTime === 9223372036854776000) {
+            expiryTimeString = "9999-12-31 23:59:59";
+          } else {
+            const expiryTimeDate = new Date(expiryTime * 1000); // requires milliseconds
+            expiryTimeString = moment(expiryTimeDate).format(
+              "YYYY-MM-DD HH:mm:ss",
+            );
+          }
+          update.expiry = expiryTimeString;
+          update.is_http_only = boolToInt(cookie.httpOnly);
+          update.is_session = boolToInt(cookie.session);
 
-        loggingDB.saveRecord("javascript_cookies", update);
-      }
-    },
-    true,
-  );
-};
+          // Accessed time (in microseconds)
+          // const lastAccessedTime = new Date(cookie.lastAccessed / 1000); // requires milliseconds
+          // update.last_accessed = lastAccessedTime.toLocaleFormat("%Y-%m-%d %H:%M:%S");
+          // update.raw_host = escapeString(cookie.rawHost);
+
+          // update.expires = cookie.expires;
```

Comment:
Let's grab all properties that are available: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/Cookie#Type with the exception of `storeId`....I don't see a use for that.

Reply:
`policy` and `status` are related to the P3P policy of the cookie. I don't think there's a way to get these, but that's fine as P3P is long dead.

`hostOnly` is the negation of `isDomain` (https://searchfox.org/mozilla-central/source/toolkit/components/extensions/parent/ext-cookies.js#23)

I don't think `raw_host` is exposed.

Reply:
Removed policy, status, is_domain and expires